tests/test_main.py

```diff
@@ -337,3 +337,55 @@ class Config:
     with pytest.raises(ValueError) as exc_info:
         m.b = 11
     assert '"TestModel" object has no field "b"' in str(exc_info)
+
+
+class ValidateAssignmentModel(BaseModel):
+    a: int = 2
+    b: constr(min_length=1)
+    c: constr(min_length=a)
+
+    class Config:
+        validate_assignment = True
+
+
+def test_validating_assignment():
+    p = ValidateAssignmentModel(a=5, b='hello', c='testing')
+    p.a = 2
+    assert p.a == 2
+    assert p.values() == {'a': 2, 'b': 'hello', 'c': 'testing'}
+    p.b = 'hi'
+    assert p.b == 'hi'
+    assert p.values() == {'a': 2, 'b': 'hi', 'c': 'testing'}
+    with pytest.raises(ValidationError) as exc_info:
+        p.a = 'b'
+    assert 'error validating input' in str(exc_info)
```

Comment:
almost all `ValidationErrors` will include `'error validating input'` either remove this or check the inner error is right.